src/AppInstallerCommonCore/AppInstallerTelemetry.cpp

```diff
@@ -87,13 +87,28 @@ namespace AppInstaller::Logging
         m_isRuntimeEnabled = true;
     }
 
+    void TelemetryTraceLogger::SetUserSettingsStatus()
+    {
+        m_isSettingEnabled = !Settings::User().Get<Settings::Setting::TelemetryDisable>();
+    }
+
+    void TelemetryTraceLogger::SetTelemetryCorelationJson(std::wstring jsonStr)
```

Comment:
This is going to need to verify that the data is JSON and wrap it if it isn't.

Reply:
I would also try to understand what type tracelogging really wants and store it as such (if it is converting to UTF-8 internally, we should store it that way to reduce the number of conversions).

Reply:
Isn't our contract with COM caller that they will pass JSON. Isn't it caller's responsibility to ensure it is a JSON string ?

Reply:
We can mention the type explicitly like in this example below. I checked few examples internally in Windows too.
                TraceLoggingPackedFieldEx(
                    m_telemetryCorelationJson.c_str(),
                    static_cast<ULONG>(m_telemetryCorelationJson.length() + 1) * sizeof(wchar_t),
                    **TlgInUNICODESTRING,**
                    TlgOutJSON,
                    "COMCallerCj"),

Reply:
We use UTF-8 everywhere else, and I would prefer to keep doing so here if that is the case.